### DISCUSSION TITLE
Unexpected timeouts with wsl1-bash

### DISCUSSION THREAD

**Discussion by ssbarnea** [Vampire/setup-wsl#25] [Q&A]
While I was able to successfully use wsl1-bash for some time, about two days ago I they started to always timeout on a particular tox task, even if I add `timeout-minutes` to them and enabled step debug.

Example: https://github.com/ansible-community/ansible-lint/runs/5166608277?check_suite_focus=true --- and keep in mind that that that task has a timeout of 5 minutes.

They run until the workflow timeout is met (30min) and the job is killed.

Did anyone encountered a similar kind of issue?

## Update

Apparently I observed a similarly perverse slowdown on my WSL2 based Windows test box where running tox seems to take ages with 100% cpu usage, even installing dependencies seems to never end.

This make me believe that https://stackoverflow.com/a/68974497/99834 or the use of `/mnt/c/` should be avoided at all costs when under WSL1 or WSL2, as it is so slow that is unusable.

As my use-pattern was to clone natively and symlink `/mnt/c/../repo` to `~/repo` and use that I think I will need to replace that with a file copy and probably followed up by a copy-back in case I need to use other native steps post testing, like log archival.

What is VERY strange is that started to behave like think only a couple of days ago, until that it was bit slow (like taking double time than on linux), but not as in never ending kind of slow.

**Comment by Vampire** [comment]
Sounds to me very similar to #23
